### PR TITLE
Add provenance-checker OpenShift template and Argo workflow

### PR DESCRIPTION
**Pull request by fridex**

### CODE REVIEW

File: provenance-checker/base/openshift-templates/provenanceChecker-template.yaml [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                arguments:
+                  artifacts:
+                    - name: outputdocument
+                      from: "{{tasks.provenance-check.outputs.artifacts.outputdocument}}"
+                  parameters:
+                    - name: "THOTH_DOCUMENT_ID"
+                      value: "{{workflow.parameters.THOTH_DOCUMENT_ID}}"
+                    - name: "THOTH_FORCE_SYNC"
+                      value: "{{workflow.parameters.THOTH_FORCE_SYNC}}"
+
```
Comment (harshad16) [inline provenance-checker/base/openshift-templates/provenanceChecker-template.yaml:143]: the extra newline is causing the pre-commit failure.

File: provenance-checker/base/argo-workflows/provenance-check-template.yaml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+              insecure: true
+              accessKeySecret:
+                name: argo-artifact-repository-secrets
+                key: accessKey
+              secretKeySecret:
+                name: argo-artifact-repository-secrets
+                key: secretKey
+      container:
+        name: provenance-check
+        image: quay.io/thoth-station/adviser:v0.6.0-dev
```
Comment (pacospace) [inline provenance-checker/base/argo-workflows/provenance-check-template.yaml:48]: this should be simply `adviser` as we use images created in the namespace where the workflow runs.

File: provenance-checker/base/openshift-templates/provenanceChecker-template.yaml (hunk 1 above)
Comment (fridex) [reply provenance-checker/base/openshift-templates/provenanceChecker-template.yaml]: Should be fixed, thanks for pointing this out! :+1:

File: provenance-checker/base/argo-workflows/provenance-check-template.yaml (hunk 2 above)
Comment (fridex) [reply provenance-checker/base/argo-workflows/provenance-check-template.yaml:48]: Oh, forgot these from testing. Nice catch...

**Comment by pacospace** [comment]
What about the other files required by Argo CD? as in https://github.com/thoth-station/thoth-application/tree/master/adviser for example?

**Comment by fridex** [comment]
> What about the other files required by Argo CD? as in https://github.com/thoth-station/thoth-application/tree/master/adviser for example?

Yes, they will be required. I've added templates I had on disc to have them tracked.

**Pull request merged**